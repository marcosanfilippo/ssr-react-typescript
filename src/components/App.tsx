import * as React from 'react';

class App extends React.Component {
  render() {
    return <h1>Welcome to SSR for React with Typescript!</h1>;
  }
}

export default App;
