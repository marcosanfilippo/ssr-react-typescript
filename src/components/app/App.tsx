import * as React from 'react';

class App extends React.Component<Title> {
  static defaultProps: any;

  render() {
    return (
      <div>
        <h1>Welcome to SSR for React with Typescript!</h1>
        <p>This is: {this.props.title}</p>
      </div>
    );
  }
}

export default App;
