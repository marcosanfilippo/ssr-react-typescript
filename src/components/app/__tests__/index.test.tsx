// tslint:disable:no-implicit-dependencies
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import App from '../App';

describe('App component', () => {
  it('renders correctly with a title', () => {
    const dummyTitle = 'dummy title';
    const tree = renderer.create(<App title={dummyTitle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// take a look here for dynamic matching: 
// https://github.com/facebook/jest/blob/master/examples/snapshot/__tests__/link.react.test.js
