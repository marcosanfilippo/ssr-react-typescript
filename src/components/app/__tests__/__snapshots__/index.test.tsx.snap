// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly with a title 1`] = `
<div>
  <h1>
    Welcome to SSR for React with Typescript!
  </h1>
  <p>
    This is: 
    dummy title
  </p>
</div>
`;
